File: iteration2/src/test/CourseStatisticsTest.java
package test;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;

import CSE3063F22P1_GRP9.CourseStatistics;


class CourseStatisticsTest {
	
	private int registeredStudentCount;
	private int registrationFailureCount;
	private double rateOfSucessfullRegistraion;
	@Test
	void getRateOfSucessfullRegistraion() {
		
		
		CourseStatistics csTesting= new CourseStatistics();
		int i;
		for(i=0;i<5;i++) {
		csTesting.incrRegisteredStudentCount();
		
		}
		
		for(i=0;i<1;i++) {
			csTesting.incrRegistrationFailureCount();
			
			}
		
		registeredStudentCount=	csTesting.getRegisteredStudentCount();
		registrationFailureCount = csTesting.getRegistrationFailureCount();
		
		
		
		rateOfSucessfullRegistraion=(1.0 * registeredStudentCount)/ (registrationFailureCount + registeredStudentCount);
	    
	    
	    
		assertEquals(rateOfSucessfullRegistraion,csTesting.getRateOfSucessfullRegistraion());
		
	}
}

File: iteration3/src/CourseStatisticsTest.py
import unittest
from CSE3063F22P1_GRP9 import CourseStatistics


class CourseStatisticsTest(unittest.TestCase):
    def getRateOfSucessfullRegistraion(self):
        csTesting = CourseStatistics()

        for i in range(5):
            csTesting.incrRegisteredStudentCount()

        for i in range(1):
            csTesting.incrRegistrationFailureCount()

        registeredStudentCount = csTesting.getRegisteredStudentCount()
        registrationFailureCount = csTesting.getRegistrationFailureCount()

        rateOfSucessfullRegistraion = (1.0 * registeredStudentCount) / (
                    registrationFailureCount + registeredStudentCount)

        self.assertEqual(rateOfSucessfullRegistraion, csTesting.getRateOfSuccessfulRegistration())


if __name__ == '__main__':
    unittest.main()
